api/src/backend/api/v1/views.py

```diff
@@ -290,26 +290,19 @@ class UserViewSet(BaseUserViewset):
     filterset_class = UserFilter
     ordering = ["-date_joined"]
     ordering_fields = ["name", "email", "company_name", "date_joined", "is_active"]
+    # RBAC required permissions
     required_permissions = [Permissions.MANAGE_USERS]
-    permission_classes = BaseRLSViewSet.permission_classes + [HasPermissions]
-
-    def initial(self, request, *args, **kwargs):
-        """
-        Sets required_permissions before permissions are checked.
-        """
-        self.required_permissions = self.get_required_permissions()
-        super().initial(request, *args, **kwargs)
 
     def get_required_permissions(self):
         """
         Returns the required permissions based on the request method.
         """
         if self.action == "me":
             # No permissions required for me request
-            return []
+            self.required_permissions = []
         else:
             # Require permission for the rest of the requests
-            return [Permissions.MANAGE_USERS]
+            self.required_permissions = [Permissions.MANAGE_USERS]
```

Comment:
I believe this is not correct. This function should return something, not update a internal attribute. Why did you modify the former logic? Since you are setting the base required permissions as `required_permissions = [Permissions.MANAGE_USERS]`, I believe this function should be something like:

```python
    def get_required_permissions(self):
        """
        Returns the required permissions based on the request method.
        """
        if self.action == "me":
            # No permissions required for me request
            return []
        else:
            # Require permission for the rest of the requests
            return self.required_permissions
```

Reply:
Making this change implies modifying code and having to define the function get_required_permissions in all views, for me the interesting thing would be to change the name to set_required_permissions.

Reply:
I thought this was one of the Django viewset functions. If it is auxiliary and our own, it's fine. Maybe consider renaming it as you suggested in other comment.